app/assets/javascripts/modules/layout_store.js

```diff
@@ -40,16 +41,26 @@
           bottomPanelEnabled: status
         });
       });
-      PubSub.subscribe('disableModes', function() {
-        _this.state({
-          modesAvailable: false
-        });
+      PubSub.subscribe('disableMode', function(_, mode) {
+        var newMode = {};
+        newMode[mode] = false;
+        _this.state(newMode);
+      });
+      PubSub.subscribe('enableMode', function(_, mode) {
```

Comment:
Cool. I think having them disabled by default and updating the data, which propagates through the interface, is a good idea. Working well on my end.